python/ray/train/v2/_internal/state/state_actor.py

```diff
@@ -20,18 +23,73 @@ def __init__(self):
         # {run_id: {attempt_id: TrainRunAttempt}}
         self._run_attempts: Dict[str, Dict[str, TrainRunAttempt]] = defaultdict(dict)
 
+        self._export_logger = self._init_export_logger()
+
     def create_or_update_train_run(self, run: TrainRun) -> None:
         self._runs[run.id] = run
+        self._maybe_export_train_run(run)
 
     def create_or_update_train_run_attempt(self, run_attempt: TrainRunAttempt):
         self._run_attempts[run_attempt.run_id][run_attempt.attempt_id] = run_attempt
+        self._maybe_export_train_run_attempt(run_attempt)
 
     def get_train_runs(self) -> Dict[str, TrainRun]:
         return self._runs
 
     def get_train_run_attempts(self) -> Dict[str, Dict[str, TrainRunAttempt]]:
         return self._run_attempts
 
+    # ============================
+    # Export API
+    # ============================
+
+    def is_export_api_enabled(self) -> bool:
+        return self._export_logger is not None
+
+    def _init_export_logger(self) -> Optional[logging.Logger]:
+        # Proto schemas should be imported within the scope of TrainStateActor to
+        # prevent serialization errors.
+        from ray.core.generated.export_event_pb2 import ExportEvent
+
+        export_api_enabled = check_export_api_enabled(
+            ExportEvent.SourceType.EXPORT_TRAIN_RUN
+        )
+
+        if not export_api_enabled:
+            return None
+
+        log_directory = get_log_directory()
+        logger = None
+        try:
+            logger = get_export_event_logger(
+                ExportEvent.SourceType.EXPORT_TRAIN_RUN,
+                log_directory,
+            )
```

Comment:
This first param determines the filename of the `.log` file that will be written to. Both `TRAIN_RUN` and `TRAIN_RUN_ATTEMPT` source types will be written in this file and will need to be parsed properly by the reader.

@nikitavemuri  The event logger allows this but logs an error: https://github.com/ray-project/ray/blob/de2b4211fdebcb092d68c37d9720aea5ca4b6559/python/ray/_private/event/export_event_logger.py#L72-L77

I think what we want is for the filename to be `event_train_state.log` (rather than just picking one of the source type names arbitrarily) and multiple schemas being written to the file (and their source type is already attached).

Reply:
Also, a follow-up item is to add a PID somewhere in the filename to prevent collisions if there are multiple Train controllers on the same node (ex: usage with Ray Tune). Then, add wildcard logic on the reader side.

Reply:
I think for that we would need `ExportEventLoggerAdapter` to take a difference source enum that refers to the filename, rather than `ExportEvent.SourceType`. We could then only print this error if the mapping between `ExportEventLoggerAdapter.source` and `event.source_type` is not what we expect

Reply:
@matthewdeng We discussed offline and decided:
* The `get_export_event_logger` method should take in a different enum than the `SourceType.EXPORT_TRAIN_RUN`. For example, it should take in `TRAIN_STATE` and write files to `event_TRAIN_STATE.log`.
* We should also have a map of `TRAIN_STATE`: [EXPORT_TRAIN_RUN, EXPORT_TRAIN_RUN_ATTEMPT]` to indicate which source types can be written to the file. If you try to write a source type outside of this list, log this error. This is to prevent random data being written in random files.

Reply:
Should the new source enum be taken in as input or should the mapping just be handled within? 

e.g.

```python
get_export_event_logger(TRAIN_STATE)
```

vs.

```python
get_export_event_logger(EXPORT_TRAIN_RUN)
-> maps to TRAIN_STATE inside
```

Reply:
@nikitavemuri suggested the first option. And the TRAIN_STATE enum can be a python object and does not need to be included in any proto file.

Reply:
Updated PTAL